tests/test_tempered_smc.py

```diff
@@ -0,0 +1,91 @@
+"""Test the tempered SMC steps and routine"""
+import functools as ft
+
+import jax
+import jax.numpy as jnp
+import jax.scipy.stats as stats
+import numpy as np
+import pytest
+
+import blackjax.hmc as hmc
+import blackjax.tempered_smc as tempered_smc
+
+
+def potential_fn(scale, coefs, preds, x):
+    """Linear regression"""
+    logpdf = 0
+    logpdf += stats.expon.logpdf(scale, 1, 1)
+    logpdf += stats.norm.logpdf(coefs, 3 * jnp.ones(x.shape[-1]), 2)
+    y = jnp.dot(x, coefs)
+    logpdf += stats.norm.logpdf(preds, y, scale)
+    return -jnp.sum(logpdf)
+
+
+def inference_loop(rng_key, kernel, initial_state, delta_solver):
+    def cond(carry):
+        _, state, _ = carry
+        return 1 - state.lamda > 1e-6
+
+    def body(carry):
```

Comment:
Actually, how about computing the tempered potential in this loop as well so the SMC kernel above is very general? In which case you could move it to `inference/smc/kernel.py`.

The general SMC state would reduce to the MCMC state and here you could create a TemperedSMC state that includes the SMC state and delta.

Reply:
Yeah I was thinking about adding a level in the module hierarchy too to be able to split responsibilities.
Plan was indeed to have the delta calculation inside the tempering, but I put it outside as you wanted to see what it would be then. More than happy to move it back inside.

Reply:
The code structure would look something like

```python
def tempered_smc(...):
    # all the tempering logic here
    particles, info = general_smc_kernel(...)
    return particles, info
```